packages/presentation/lib/ui/auth_page/bloc/auth_bloc.dart

```diff
@@ -36,31 +39,83 @@ abstract class AuthBloc extends Bloc<BaseArguments, AuthTile> {
   Future<void> authFacebook();
 
   Future<void> authGoogle();
+
+  GlobalKey<FormState> get formKey;
+
+  String? validatorLogin(BuildContext context);
+
+  String? validatorPassword(BuildContext context);
 }
 
 class AuthBlocImpl extends BlocImpl<BaseArguments, AuthTile>
     implements AuthBloc {
   var _tile = AuthTile.init();
+  final _formKey = GlobalKey<FormState>();
   final _loginController = TextEditingController();
   final _passwordController = TextEditingController();
   final LoginGoogleUseCase loginGoogleUseCase;
   final LoginFacebookUseCase loginFacebookUseCase;
   final LoginEmailAndPassUseCase authUseCase;
+  final ValidatorUseCase validatorUseCase;
   final AnalyticsUseCase analytics;
 
+  @override
+  GlobalKey<FormState> get formKey => _formKey;
+
   @override
   TextEditingController get textLoginController => _loginController;
 
   @override
   TextEditingController get textPasswordController => _passwordController;
 
+  UserEmailPass get _enteredUser => UserEmailPass(
+        _loginController.text,
+        _passwordController.text,
+      );
+
   AuthBlocImpl({
+    required this.validatorUseCase,
     required this.authUseCase,
     required this.analytics,
     required this.loginGoogleUseCase,
     required this.loginFacebookUseCase,
   });
 
+  @override
+  String? validatorLogin(BuildContext context) {
+    final useCase = validatorUseCase(_enteredUser);
```

Comment:
usecase and has all the validation logic, two methods are written, since we have two forms and the same string will be returned for two fields, so we cannot throw one method into 2 fields